### ISSUE TITLE
FMRelationSlot should use setters instead of #instVarAt:put:

### ISSUE THREAD

**Issue by jecisc** [moosetechnology/Famix#111]
FMRelationSlot is updating the opposite slot if it exists. But, to do that, it uses `instVarAt:put:`. 

This is a problem because we might have some behavior in the setter that needs to be done.

Instead, we should use #perform:with:

To reproduce the problem it is simple. Create an entity TWithTypes and an entity TType. 

Then you add the second one to the first one using #addType: and ask the mooseName. It is not reset. This should be added as test.

**Comment by jecisc** [comment]
We could avoid the loop by using the new FMShouldCheckForDuplicatedEntitiesInMultivalueLinks variable I guess

**Comment by NicolasAnquetil** [comment]
Yes I think it's using #instVarAt:put: to avoid the opposite setting the opposite setting the opposite setting the opposite...

I don't know about FMShouldCheckForDuplicatedEntitiesInMultivalueLinks, but the behavior should work for "single" and "multiple" relationships.
So this class does not look like the right solution (given its name)

**Comment by jecisc** [comment]
Yes indeed I mixed two mechanism of the slots. 

Maybe we could add a mechanism to disable the setting of the opposite during the execution of a bloc as I did for the duplication check

**Comment by AyechiNour** [comment]
we propose to close this issue
